### PR TITLE
feat: [#257] add logic for picking arrow functions body

### LINKED ISSUE #257: Styled component interpolation function behaviour
See: [https://github.com/compiled/compiled-css-in-js/blob/0954b67b19530685d4505a0570e2404eeb78193c/packages/babel-plugin/src/styled/__tests__/index.test.tsx#L111](https://github.com/compiled/compiled-css-in-js/blob/0954b67b19530685d4505a0570e2404eeb78193c/packages/babel-plugin/src/styled/__tests__/index.test.tsx#L111)

Short circuiting is already implemented - what we need to do is update the code so we pick out the arrow functions body. Right now the output is:

`"--var-hash-test\":((props=>props.color)||\"\")`

It should be:

`"--var-hash-test\":(props.color||\"\")`

Think about other tests to ensure this implementation works.

This implementation is also partially implemented here: [https://github.com/compiled/compiled-css-in-js/blob/023d2f5afefa659a7ab0787af1ea0ab648f3a89f/packages/babel-plugin/src/utils/ast-builders.tsx#L136](https://github.com/compiled/compiled-css-in-js/blob/023d2f5afefa659a7ab0787af1ea0ab648f3a89f/packages/babel-plugin/src/utils/ast-builders.tsx#L136)

We want to update the code so it matches more use cases/scenarios.

We should also handle the case when the arrow function has a body with brackets, see test here: [https://github.com/compiled/compiled-css-in-js/blob/0954b67b19530685d4505a0570e2404eeb78193c/packages/babel-plugin/src/styled/__tests__/index.test.tsx#L333](https://github.com/compiled/compiled-css-in-js/blob/0954b67b19530685d4505a0570e2404eeb78193c/packages/babel-plugin/src/styled/__tests__/index.test.tsx#L333)

```typescript
styled.div`
  color: ${props => props.color};
  color: ${props => { return props.color }};
`;
```

We aren’t worrying about function declarations, e.g. `function () {}`

Update styled ast function to handle the above scenarios: [https://github.com/compiled/compiled-css-in-js/blob/023d2f5afefa659a7ab0787af1ea0ab648f3a89f/packages/babel-plugin/src/utils/ast-builders.tsx#L136](https://github.com/compiled/compiled-css-in-js/blob/023d2f5afefa659a7ab0787af1ea0ab648f3a89f/packages/babel-plugin/src/utils/ast-builders.tsx#L136)

**Testing:**

A few of these tests will be able to just enable and they will pass after fixing.

For the most part the wiring up of this is done - it’s a matter of uncommenting tests and seeing what needs to be changed.

### PR DESCRIPTION
**Pull request by pgmanutd**
closes #257

### CODE REVIEW

File: packages/babel-plugin/src/styled/__tests__/index.test.tsx [hunk 1]
```diff
@@ -339,8 +339,51 @@ describe('styled component transformer', () => {
         \`;
       `);
 
-      expect(actual).toInclude('.css-test{color:var(--var-test-propscolor)}');
-      expect(actual).toInclude('"--var-test-propscolor": (() => { return props.color; })() }}');
+      expect(actual).toInclude('.cc-hash-test{color:var(--var-hash-test)}');
+      expect(actual).toInclude('"--var-hash-test":(()=>{return props.color;})()}}');
+    });
+
+    it('should transform an arrow function with a body into an IIFE by preventing passing down invalid html attributes to the node', () => {
```
Comment (madou) [inline packages/babel-plugin/src/styled/__tests__/index.test.tsx:346]: great addition

File: packages/babel-plugin/src/utils/ast-builders.tsx [hunk 2]
```diff
@@ -76,6 +76,49 @@ const styledStyleProp = (
...
+ *
+ * @param node Node of type either BlockStatement or Expression
+ */
+export const wrapNodeInIIFE = (node: t.BlockStatement | t.Expression) =>
+  t.callExpression(t.arrowFunctionExpression([], node), []);
+
+const tryWrappingBlockStatementInIIFE = (node: t.BlockStatement | t.Expression) =>
+  t.isBlockStatement(node) ? wrapNodeInIIFE(node) : node;
+
+const pickArrowFunctionExpressionBody = (node: t.ArrowFunctionExpression) => {
```
Comment (madou) [inline packages/babel-plugin/src/utils/ast-builders.tsx:91]: nit: we can move this into `ast.tsx` when my PR lands

**Review by madou — APPROVED** [review]
would be great for us to add some storybook examples to make sure theory and practice are aligning. if you have 5 minutes today let's add some before merging?

**Comment by pgmanutd** [comment]
Yups sure. Thanks. Will just add few examples 😄

**Review by madou — APPROVED** [review]

**Pull request merged**